### ISSUE TITLE
Keep measured line(s) visible and make snapable during editing

### ISSUE THREAD

**Issue by beyhh** [qgis/QGIS#37059]
Any line which is measured with the Measure Tool is disappearing as soon as the User starts editing a certain layer (like adding or modifying a feature). 
Instead, the measured line should stay visible and should even be snapable during the editing process. This way, the user can measure off a distance to a certain object or landmark (on the background chart in form of WMS or XYZ tile) and determine the position where the feature need to be drawn.

**Comment by gioman** [comment]
@beyhh have you seen the advanced digitizing panel? it gives you the distance (and other measures) while you draw lines/segments.

**Comment by beyhh** [comment]
Yes, thanks for pointing this out. However, if I would like to measure the distance from anything which I see on the "background chart" (WMS or XYZ tiles), how can I do this with Advanced Digitizing? Also, how can I change the unit of the distance measurement in Advanced Digitizing? I would like to have the option for different units but can't see how this can be done. All this would be possible with the Measurement Tool, as long as it stays available during editing.

**Comment by beyhh** [comment]
I just learned that the _Shape Tools_ Plugin has a _Geodesic Measure Tool_ which does exactly what I was looking for: it stays visible and does not disappear when editing/drawing on a different layer. This tool is very similar to QGIS' built-in version and the idea is to make the Geodesic Measure Tool the standard measure tool and replace the angle measure tool as well. What are your thoughts on this? I would only like to have the option to set the angle measurement to 000° - 360° notation as currently angles are only measured from -180° to 180°.
